dotCMS/src/main/java/com/dotcms/util/pagination/TemplatePaginator.java

```diff
@@ -0,0 +1,107 @@
+
+/**
+ * Handle {@link com.dotmarketing.portlets.templates.model.Template} pagination
+ * @author jsanca
+ */
+public class TemplatePaginator implements PaginatorOrdered<TemplateView> {
+
+    public static final String HOST_PARAMETER_ID = "host";
+
+    private final TemplateAPI templateAPI;
+    private final TemplateHelper templateHelper;
+
+    public TemplatePaginator() {
+        this(APILocator.getTemplateAPI(),
+                new TemplateHelper(APILocator.getPermissionAPI(), APILocator.getRoleAPI()));
+    }
+
+    @VisibleForTesting
+    public TemplatePaginator(final TemplateAPI templateAPI, final TemplateHelper templateHelper) {
+
+        this.templateAPI    = templateAPI;
+        this.templateHelper = templateHelper;
+    }
+
+    @Override
+    public PaginatedArrayList<TemplateView> getItems(final User user, final String filter, final int limit, final int offset,
+                                                  final String orderby, final OrderDirection direction,
+                                                  final Map<String, Object> extraParams) {
+        String hostId = null;
+
+        if (extraParams != null) {
+            hostId = (String) extraParams.get(HOST_PARAMETER_ID);
+        }
+
+        final Map<String, Object> params = map("title", filter);
+
+        String orderByDirection = orderby;
+        if (UtilMethods.isSet(direction) && UtilMethods.isSet(orderby)) {
+            orderByDirection = new StringBuffer(orderByDirection)
+                    .append(" ")
+                    .append(direction.toString().toLowerCase()).toString();
+        }
+
+        try {
+            final PaginatedArrayList<Template> allTemplates =
+                    (PaginatedArrayList<Template>) templateAPI.findTemplates(user, false, params, hostId,
+                    null, null, null, offset, limit, orderByDirection);
+
+            if (UtilMethods.isSet(hostId)) {
+
+                allTemplates.stream().sorted(
+                        direction == OrderDirection.ASC?Comparator.comparing(this::hostname):
+                                Comparator.comparing(this::hostname).reversed());
+            }
+
+            final PaginatedArrayList<TemplateView> templates =
+                    new PaginatedArrayList<>();
+            templates.addAll(allTemplates.stream().map(template -> this.templateHelper.toTemplateView(template, user)).collect(Collectors.toList()));
+            templates.setQuery(allTemplates.getQuery());
+            templates.setTotalResults(allTemplates.getTotalResults());
+
+            return templates;
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new PaginationException(e);
+        }
+    }
+
+    private String hostname (final Template template) {
+
+        try {
+            return Host.class.cast(template.getParentPermissionable()).getHostname();
+        } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6042990)